qiskit/transpiler/synthesis/graysynth.py

```diff
@@ -241,6 +243,13 @@ def _lwr_cnot_synth(state, section_size):
     Patel, Ketan N., Igor L. Markov, and John P. Hayes.
     Quantum Information & Computation 8.3 (2008): 282-294.
 
+    Matt's tweak:
+    To get decent performance out of PMH you need to play around with it a bit.
+    This version adds a simple "back reduce" that lets rows below the pivot
+    which have significant overlap with the pivot to be added back to the pivot.
+    The intuition is to avoid the situation where a high-weight pivot row is
+    propagated to other rows. Seems to work well in practice.
+
```

Comment:
Is this note meant to be for the user? This is an internal method. In any case, it is seems very ambiguous to me. What do you mean by "play around with it a bit"?

Reply:
The main reason was to explain the bug and the change that fixes it, and to leave the explanation there for posterity/reproducibility since the docstring also explains the PMH algorithm. I think it's a bit less ambiguous now. Feel free to remove in any case.

One thing to note, it doesn't necessarily produce better CNOT counts, but I've found it does better than straight PMH on average. Hence the "play around with" --- I use a couple of strategies and pick the best result in most cases.

Reply:
much better! thanks!